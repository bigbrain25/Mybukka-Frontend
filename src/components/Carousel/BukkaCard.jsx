import React from 'react';

import PropTypes from 'prop-types';

import MarkIcon from 'Icons/Remark';
import Navlink from 'Components/navlink/Navlink';
import Price from 'Components/badge/Price';

import './BukkaCard.scss';

const TextOverlay = () => (
  <div className="overlay">
    <h4>Chipotle</h4>
    <h5>Order Now</h5>
  </div>
);

const NormalText = ({ deliveryCost, deliveryTime, rating }) => (
  <div className="details">
    <h3>
      <span className="details-location">
        San francisco
        <span>
          kitchen
          <span className="icon">
            <MarkIcon />
          </span>
        </span>
      </span>
      {rating && <span className="rating">{rating}</span>}
    </h3>
    <div className="delivery">
      {deliveryCost ? <Price price={deliveryCost} /> : null}
      {deliveryCost ? <i className="dot-spacing">.</i> : null}
      <div className="delivery-time-text">{deliveryTime}</div>
    </div>
  </div>
);

const BukkaCard = ({
  imageUrl,
  deliveryCost,
  deliveryTime,
  rating,
  imageHeight,
  textOverlay
}) => (
  <div className={['mt-4 bukka-card'].join(' ')}>
    <img className={`bukka-img ${imageHeight}`} src={imageUrl} alt="alt_image" />
    {textOverlay && <TextOverlay />}
    {!textOverlay && (
      <NormalText
        deliveryCost={deliveryCost}
        deliveryTime={deliveryTime}
        rating={rating}
      />
    )}
  </div>
);

const GetBukka = ({ classNames, ...props }) => (
  <div className={`card-container ${classNames}`}>
    <div className="card-wrap">
      <Navlink classNames="link" href="/">
        <BukkaCard {...props} />
      </Navlink>
    </div>
  </div>
);

export default GetBukka;

NormalText.defaultProps = {
  deliveryTime: '20 - 40',
  rating: '',
  deliveryCost: 0
};

NormalText.propTypes = {
  deliveryCost: PropTypes.number,
  deliveryTime: PropTypes.string,
  rating: PropTypes.string
};

BukkaCard.defaultProps = {
  deliveryTime: '20 - 40',
  rating: '',
  imageHeight: '',
  textOverlay: false,
  deliveryCost: 0,
};

BukkaCard.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  deliveryCost: PropTypes.number,
  deliveryTime: PropTypes.string,
  rating: PropTypes.string,
  imageHeight: PropTypes.string,
  textOverlay: PropTypes.bool
};

GetBukka.defaultProps = {
  classNames: ''
};

GetBukka.propTypes = {
  classNames: PropTypes.string
};
