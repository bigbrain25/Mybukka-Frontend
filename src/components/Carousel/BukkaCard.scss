@import '../../stylebase/variables.scss';
$fontFamily: 'Roboto', sans-serif;

.card-container {
  position: relative;
  padding-bottom: 20px;
}

.default-overlay {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  top: 10px;
  right: 12px;
  left: 0;
}

.bukka-img-color {
  background: rgb(233, 231, 231);
}

.card-wrap {
  display: flex;
  flex-shrink: 0;
  flex-direction: column;
  cursor: pointer;
  position: relative;
}

.bukka-card {
  position: relative;
  background-color: transparent;
  display: flex;
  overflow: hidden;
}

.touchdown {
  .rating {
    display: none;
  }
}

.details {
  position: relative;
  padding-top: 16px;
}

.details h3 {
  display: flex;
  margin-top: 0px;
  margin-bottom: 4px;
  font-family: $fontFamily;
  font-weight: 500;
  font-size: 16px;
  text-transform: capitalize;
  @media (max-width: 767px) {
    font-size: 13px;
  }
}

.collection-image-spacing {
  height: auto;
  padding-bottom: 58.82352941176471%;
}

.normal-image-spacing {
  height: auto;
  padding-bottom: 61.1%;
}

.cuisine-image-spacing {
  height: auto;
  padding-bottom: 119.04761904761905%;
}

.category-image-spacing {
  height: auto;
  padding-bottom: 105.047619%;
}

.custom-bukka-image {
  position: absolute;
  background-position: center center;
  background-size: cover;
  background-repeat: no-repeat;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-transition: opacity 0.4s linear;
  transition: opacity 0.4s linear;
  background-color: rgb(233, 231, 231);
}


.custom-delivery-badge {
  color: #9c1c26;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0.72px;
  position: relative;
  top: -4px;
  padding-left: 5px;
}

.delivery {
  display: flex;
  flex-direction: row;
  font-family: $fontFamily;
  font-size: 14px;
  letter-spacing: 0.14px;
  font-weight: 400;
  line-height: 24px;
  margin-top: 10px;
  color: #8f95a3;
}

.bukka-card:hover .favorite {
  opacity: 1;
}

.other-bukka-card {
  padding-top: 69px;
}

.favorite {
  opacity: 0;
  transition: opacity 200ms ease-in;
}

.delivery-fav {
  display: inline-block;
  margin-right: 5px;
  color: gold;
}

.delivery-time-text
 {
  margin-top: -3px;
  margin-left: 4px;
}

.dot-spacing {
  margin-top: -7px;
  margin-left: 4px;
}

.details-location {
  flex: 1;
  color: $color-dark-gray;
  word-break: break-all;
}

.rating {
  background-color: rgba(0, 198, 208, 0.1);
  display: inline-flex;
  color: #00c6d0;
  font-size: 11px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.6px;
  padding: 0 14px;
  align-items: center;
  height: 24px;
  margin: -3px 0 -3px 8px;
}

.icon {
  align-self: center;
  height: 14px;
  width: 14px;
  margin-left: 8px;
  vertical-align: -2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

.overlay-bottom, .overlay-top {
  margin: 24px 24px 0;
  max-width: 164px;
}

.overlay-bottom {
  position: absolute;
  bottom: 24px;
}

.overlay-top {
  position: absolute;
  top: 0;
}

.overlay-h4s {
  font-size: 18px;
  font-weight: 600;
  text-transform: capitalize;
  line-height: normal;
  letter-spacing: -0.2px;
}

.overlay-h4, .overlay-top h4, .overlay-top h5, .overlay-bottom h5
{
  font-family: 'Roboto', sans-serif;
}

.overlay-top h4 {
  text-transform: capitalize
}

.overlay-h4, .overlay-top h4 {
  color: black;
  font-size: 24px;
}

.overlay-top h5 {
  color: black;
  font-size: 16px;
  @media (max-width: 992px) {
    font-size: 14px
  }
  @media (max-width: 767px) {
    font-size: 14px
  }
}

.overlay-bottom h5 {
  color: gray;
  font-size: 13px
}

@media screen and (min-width: 768px) {
  .overlay-bottom, .overlay-top {
    max-width: 190px;
  }
}

@media screen and (max-width: 444px) {
  .overlay-bottom, .overlay-top {
    margin-left: 15px;
    margin-top: 10px
  }

  .overlay-h4, .overlay-top h4 {
    font-size: 16px;
  }

  .overlay-top h5, .overlay-bottom h5 {
    font-size: 10px
  }
}

.img-small-screen {
  display: block;
  @media screen and (min-width: 576px) {
    display: none
  }
}

.img-large-screen {
  display: block;
  @media screen and (max-width: 576px) {
    display: none
  }
}
