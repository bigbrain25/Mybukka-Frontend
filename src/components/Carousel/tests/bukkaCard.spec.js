import React from 'react';
import BukkaCard from '../BukkaCard';

describe('BukkaCard component', () => {
  const props = {
    deliveryCost: 1000,
    deliveryTime: '20-20 hours',
    rating: '',
    classNames: 'col-lg-3',
    imageHeight: 'img-height',
    textOverlay: true,
    imageUrl: 'http://some-image',
  };
  const wrapper = shallow(<BukkaCard {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
