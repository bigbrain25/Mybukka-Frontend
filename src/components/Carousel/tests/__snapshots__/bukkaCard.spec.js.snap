// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BukkaCard component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetBukka
    classNames="col-lg-3"
    deliveryCost={1000}
    deliveryTime="20-20 hours"
    imageHeight="img-height"
    imageUrl="http://some-image"
    rating=""
    textOverlay={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-wrap"
      >
        <Navlink
          classNames="link"
          href="/"
          text=""
        >
          <BukkaCard
            deliveryCost={1000}
            deliveryTime="20-20 hours"
            imageHeight="img-height"
            imageUrl="http://some-image"
            rating=""
            textOverlay={true}
          />
        </Navlink>
      </div>,
      "className": "card-container col-lg-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navlink
          classNames="link"
          href="/"
          text=""
        >
          <BukkaCard
            deliveryCost={1000}
            deliveryTime="20-20 hours"
            imageHeight="img-height"
            imageUrl="http://some-image"
            rating=""
            textOverlay={true}
          />
        </Navlink>,
        "className": "card-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <BukkaCard
            deliveryCost={1000}
            deliveryTime="20-20 hours"
            imageHeight="img-height"
            imageUrl="http://some-image"
            rating=""
            textOverlay={true}
          />,
          "classNames": "link",
          "href": "/",
          "text": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "deliveryCost": 1000,
            "deliveryTime": "20-20 hours",
            "imageHeight": "img-height",
            "imageUrl": "http://some-image",
            "rating": "",
            "textOverlay": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-wrap"
        >
          <Navlink
            classNames="link"
            href="/"
            text=""
          >
            <BukkaCard
              deliveryCost={1000}
              deliveryTime="20-20 hours"
              imageHeight="img-height"
              imageUrl="http://some-image"
              rating=""
              textOverlay={true}
            />
          </Navlink>
        </div>,
        "className": "card-container col-lg-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Navlink
            classNames="link"
            href="/"
            text=""
          >
            <BukkaCard
              deliveryCost={1000}
              deliveryTime="20-20 hours"
              imageHeight="img-height"
              imageUrl="http://some-image"
              rating=""
              textOverlay={true}
            />
          </Navlink>,
          "className": "card-wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <BukkaCard
              deliveryCost={1000}
              deliveryTime="20-20 hours"
              imageHeight="img-height"
              imageUrl="http://some-image"
              rating=""
              textOverlay={true}
            />,
            "classNames": "link",
            "href": "/",
            "text": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "deliveryCost": 1000,
              "deliveryTime": "20-20 hours",
              "imageHeight": "img-height",
              "imageUrl": "http://some-image",
              "rating": "",
              "textOverlay": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
